sdk/typespec/typespec_derive/src/lib.rs

```diff
@@ -0,0 +1,205 @@
+extern crate proc_macro;
+
+use proc_macro2::{Span, TokenStream};
+use syn::parse::ParseStream;
+use syn::spanned::Spanned;
+use syn::{Attribute, DeriveInput, LitStr, Meta, Path};
+
+/// Derive macro for implementing `Model` trait.
+/// Deriving this trait allows a type to be deserialized from an HTTP response body.
+/// By default, the type must also implement `serde::Deserialize`, or the generated code will not compile.
+///
+/// ## Attributes
+///
+/// The following attributes are supported on the struct itself:
+///
+/// ### `#[typespec(crate)]`
+///
+/// The 'crate' attribute specifies an alternate module path, other than the default of `typespec_client_core`, to reference the typespec client crate.
+///
+/// ```rust
+/// # use typespec_derive::Model;
+/// # use serde::Deserialize;
+/// extern crate typespec_client_core as my_typespec;
+///
+/// #[derive(Model, Deserialize)]
+/// #[typespec(crate = "my_typespec")]
+/// struct MyModel {
+///   value: String
+/// }
+/// ```
```

Comment:
Nit: I expect this will get used less. I imagine you're alphabetizing - I know I like to most times :smile: - but I'd mention the `format` first since it's likely going to get used more...and still probably not that much.

Reply:
Yep, I agree. I wasn't even conciously alphabetizing, I just had to implement `crate` first to get tests to compile before I even started on `format`. I agree that `format` should come first.